collectors/googlestackdriver/collector.js

```diff
@@ -0,0 +1,120 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2019, Alert Logic, Inc
+ * @doc
+ *
+ * googlestackdriver class.
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+'use strict';
+
+const moment = require('moment');
+const PawsCollector = require('@alertlogic/paws-collector').PawsCollector;
+const parse = require('@alertlogic/al-collector-js').Parse;
+const logging = require('@google-cloud/logging');
+
+
+const typeIdPaths = [
+    {path: ['protoPayload', 'type_url']}
+];
+
+class GooglestackdriverCollector extends PawsCollector {
+    constructor(context, creds) {
+        super(context, creds, 'googlestackdriver');
+    }
+    
+    pawsInitCollectionState(event, callback) {
+        // TODO: put in some more efficient catching up logic for historical logs. Stackdriver stores logs for 30 days
+        const startTs = process.env.paws_collection_start_ts ? 
+                process.env.paws_collection_start_ts :
+                    moment().toISOString();
+        const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+        const initialState = {
+            since: startTs,
+            until: endTs,
+            poll_interval_sec: 1
+        };
+        return callback(null, initialState, 1);
+    }
+    
+    pawsGetLogs(state, callback) {
+        let collector = this;
+
+        // Start API client
+        const client = new logging.v2.LoggingServiceV2Client({
+            credentials: JSON.parse(this._pawsCreds.secret)
+        });
+
+
+        console.info(`GSTA000001 Collecting data from ${state.since} till ${state.until}`);
+
+        // TODO: figure out a better way to format this. I'm pretty sure that it needs the newlines in it.
+        const filter = `timestamp >= "${state.since}"
+timestamp < "${state.until}"`;
+        const formattedResourceNames = JSON.parse(process.env.google_project_ids).map(e => `projects/${e}`);
```

Comment:
Is it possible to specify `all` projects?

Reply:
I don't think that's possible to per this api doc. https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/list

Thinking about it, it wouldnt make sense. like i don't think you can collector logs form every vpc in AWS or every rg in Azure.

You can specify an organization, which can sit above several projects. I think that will collect logs from several projects, but I have not validated that idea.

Reply:
In Azure it is possible to get activity logs for all RGs. Should be something similar for collection all logs here.
Maybe here we can use `projects/*`.
Actually I think we can replace `google_project_ids` with `google_resource_names` and simply pass this parameter into `client.listLogEntries()` as is. Thoughts?

Reply:
So in my own tests with teh api explorer, it does not accept `projects/*`

But yeah we could make that parameter resource names, but then they would need to be in the format `<resource-type>/<name>` in the param. I just restricted to project because that was 1. what i could test an 2. seemed to be the main 'unit of organization' for google cloud.

Reply:
I'd still stick to `google_resource_names` because logs can be retrieved for multiple resource types.
> "projects/[PROJECT_ID]"
"organizations/[ORGANIZATION_ID]"
"billingAccounts/[BILLING_ACCOUNT_ID]"
"folders/[FOLDER_ID]"

Reply:
Thats fair. I'll change it then.